drivers/accel/adxl313/iio_adxl313.c

```diff
@@ -0,0 +1,1146 @@
+				   uint32_t len, const struct iio_ch_info *channel, intptr_t priv)
+{
+	int32_t vals[2];
+	int ret = 0;
+	struct adxl313_iio_dev *iio_adxl313;
+	struct adxl313_dev *adxl313;
+
+	if (!dev)
+		return -EINVAL;
+
+	iio_adxl313 = (struct adxl313_iio_dev *)dev;
+
+	if (!iio_adxl313->adxl313_dev)
+		return -EINVAL;
+
+	adxl313 = iio_adxl313->adxl313_dev;
+
+	if (channel->type == IIO_ACCEL) {
+		switch (adxl313->dev_type) {
+		case ID_ADXL312:
+			iio_parse_value(buf, IIO_VAL_FRACTIONAL, &vals[0], &vals[1]);
+			/* Search for the given data in the range available data
+			 * Enable/disable FULL_RES depending on selection */
+			if (vals[1] == adxl312_iio_scale_table[0])
+				return adxl313_enable_full_res(adxl313);
+			else
+				return adxl313_disable_full_res(adxl313);
+		case ID_ADXL313:
+			iio_parse_value(buf, IIO_VAL_FRACTIONAL, &vals[0], &vals[1]);
+			/* Enable/disable FULL_RES depending on selection */
+			if (vals[1] == adxl313_iio_scale_table[0])
+				return adxl313_enable_full_res(adxl313);
+			else
+				return adxl313_disable_full_res(adxl313);
+		case ID_ADXL314:
+			return ret;
+		default:
+			return -EINVAL;
+		}
+	}
+
+	return -EINVAL;
+}
+
+/*******************************************************************************
+ * @brief Handles the read request for calibbias attribute.
+ *
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ *
+ * @return ret    - Result of the reading procedure.
+ * 					In case of success, the size of the read data is returned.
+*******************************************************************************/
+static int adxl313_iio_read_calibbias(void *dev, char *buf, uint32_t len,
+				      const struct iio_ch_info *channel, intptr_t priv)
+{
+	int32_t val;
+	uint32_t multiplier = ADXL312_OFFSET_SCALE_FACTOR;
+	struct adxl313_iio_dev *iio_adxl313;
+	struct adxl313_dev *adxl313;
+
+	if (!dev)
+		return -EINVAL;
+
+	iio_adxl313 = (struct adxl313_iio_dev *)dev;
+
+	if (!iio_adxl313->adxl313_dev)
+		return -EINVAL;
+
+	adxl313 = iio_adxl313->adxl313_dev;
+
+	switch (adxl313->dev_type) {
+	case ID_ADXL312:
+		multiplier = ADXL312_OFFSET_SCALE_FACTOR;
+		break;
+	case ID_ADXL313:
+		multiplier = ADXL313_OFFSET_SCALE_FACTOR;
+		break;
+	case ID_ADXL314:
+		multiplier = ADXL314_OFFSET_SCALE_FACTOR;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	if (channel->type == IIO_ACCEL) {
+		switch (channel->address) {
+		case ACCEL_AXIS_X:
+			val = adxl313->x_offset_raw * multiplier;
+			return iio_format_value(buf, len, IIO_VAL_INT, 1, &val);
+		case ACCEL_AXIS_Y:
+			val = adxl313->y_offset_raw * multiplier;
+			return iio_format_value(buf, len, IIO_VAL_INT, 1, &val);
+		case ACCEL_AXIS_Z:
+			val = adxl313->z_offset_raw * multiplier;
+			return iio_format_value(buf, len, IIO_VAL_INT, 1, &val);
```

Comment:
You can have only 1 return after the `switch`, which will simplify the code a bit.